File: src/JsonValidate.JSInterop/CodeMirror.cs
using System.Threading.Tasks;
using Microsoft.JSInterop;

namespace JsonValidate.JSInterop
{
    public class CodeMirror
    {
        public static ValueTask CreateAsync(IJSRuntime jsRuntime, object textArea) =>
            jsRuntime.InvokeVoidAsync(
                "codemirror.create",
                textArea);

        public static ValueTask ClearAsync(IJSRuntime jsRuntime) =>
            jsRuntime.InvokeVoidAsync(
                "codemirror.clear");

        public static ValueTask<string> GetValueAsync(IJSRuntime jsRuntime) =>
            jsRuntime.InvokeAsync<string>(
                "codemirror.getValue");

        public static ValueTask SetValueAsync(IJSRuntime jsRuntime, string value) =>
            jsRuntime.InvokeVoidAsync(
                "codemirror.setValue",
                value);

        public static ValueTask SetOptionAsync(IJSRuntime jsRuntime, string key, string value) =>
            jsRuntime.InvokeVoidAsync(
                "codemirror.setOption",
                key,
                value);

        public static ValueTask HighlightAsync(
            IJSRuntime jsRuntime, int fromLine, int fromChar, 
            int toLine, int toChar, bool isLightTheme) =>
            jsRuntime.InvokeVoidAsync(
                "codemirror.highlight",
                fromLine,
                fromChar,
                toLine,
                toChar,
                isLightTheme);
    }
}

File: src/JsonValidate.JSInterop/wwwroot/codemirror.js
var cm;

window.codemirror = {
    create: function (textArea) {
        cm = CodeMirror.fromTextArea(textArea, {
            mode: { name: 'javascript', json: true },
            lineNumbers: true,
            tabSize: 2
        });
    },
    clear: function () {
        cm.doc.setValue('');
    },
    getValue: function () {
        var value = cm.doc.getValue();
        return value;
    },
    setValue: function (value) {
        cm.doc.setValue(value);
    },
    setOption: function (key, value) {
        cm.setOption(key, value);
    },
    highlight: function (fromLine, fromChar, toLine, toChar, isLightTheme) {
        cm.doc.markText({ line: fromLine, ch: fromChar }, { line: toLine, ch: toChar }, {
            css: `background: ${isLightTheme ? '#81dee4' : '#7E211B'}`
        })
    }
};